### DISCUSSION TITLE
TypeError: Unsupported format: <built-in function format>

### DISCUSSION THREAD

**Discussion by frankstorming** [open-mmlab/mmocr#1855] [Q&A]
I tried to convert the RCTW format,then the bug came.I can't solve it.

## Error Traceback:

It takes 13.648082733154297s to convert RCTW Training annotation
Traceback (most recent call last):
  File "/home/kzx22/ocr/mmocr/tools/dataset_converters/textrecog/rctw_converter.py", line 238, in <module>
    main()
  File "/home/kzx22/ocr/mmocr/tools/dataset_converters/textrecog/rctw_converter.py", line 227, in main
    generate_ann(root_path, 'training', trn_infos, args.preserve_vertical)
  File "/home/kzx22/ocr/mmocr/tools/dataset_converters/textrecog/rctw_converter.py", line 196, in generate_ann
    dump_ocr_data(img_info, dst_label_file, 'textrecog')
  File "/home/kzx22/ocr/mmocr/mmocr/utils/data_converter_utils.py", line 127, in dump_ocr_data
    mmengine.dump(out_json, out_json_name, **kwargs)
  File "/home/kzx22/anaconda3/envs/mmocr/lib/python3.9/site-packages/mmengine/fileio/io.py", line 908, in dump
    raise TypeError(f'Unsupported format: {file_format}')
TypeError: Unsupported format: <built-in function format>

## image_1.txt format:
2668,376,2832,376,2832,347,2668,347,0,"2号直流屏电源"
2681,402,2825,402,2825,379,2681,379,0,"4141065开关"
2657,1101,2828,1101,2828,1072,2657,1072,0,"2#UPS旁路电源"
2674,1129,2816,1129,2816,1102,2674,1102,0,"4141066开关"
637,392,772,392,772,365,637,365,0,"备用（电动机）"
643,415,772,415,772,394,643,394,0,"4141085开关"
662,442,753,442,753,421,662,421,0,"4kW8 2A"
645,1052,697,1052,697,1021,645,1021,0,"备用"
614,1081,745,1081,745,1052,614,1052,0,"4141086开关"
630,1108,724,1108,724,1081,630,1081,0,"4KW7.9A"
2680,1929,2849,1929,2849,1892,2680,1892,0,"4141067开关"
2682,1896,2841,1896,2841,1846,2682,1846,0,"EPS之电源"
508,1752,699,1752,699,1722,508,1722,0,"200#1楼上墙角风机"
541,1777,670,1777,670,1750,541,1750,0,"4141087开关"
545,1802,668,1802,668,1779,545,1779,0,"22kW42.4A"
## Finally I print the json_name
print(out_json_name):
out_json_name: data/rctw/train_label.<built-in function format>

**Comment by gaotongxiao** [comment]
https://mmocr.readthedocs.io/en/dev-1.x/basic_concepts/datasets.html#recogtextdataset

**Reply by frankstorming** [reply]
I probably get what you mean.
Space isn't allowed to exist in the .txt file.

**Reply by frankstorming** [reply]
I have removed all the space in annotations ,but I couldn't solve the problem.
By the way,I forgot to add \\ before \< and \> ,the actual value of out_json_name is as follows.
out_json_name: data/rctw/train_label.\<built-in function format\>

**Reply by frankstorming** [reply]
I traced the bug and found the out_json_name was not correct.
In 'mmocr/tools/dataset_converters/textrecog/rctw_converter.py', I found variable dst_label_file was not correct.
I replaced the string 'format' into 'json' in both line 157 and line 159 then the command work!
I wonder if there exists a bug in it.

**Reply by gaotongxiao** [reply]
Did a quick check and confirmed it's a bug when we tried to migrate the converter from 0.x to 1.x. It would be nice if you could make a patch to this script. :)

**Reply by frankstorming** [reply]
Thanks,I have never done this before,but I would try it.
It would be my honor if I could make contribution to the community.
